### PR TITLE
`FloatingPanel` - Improve keyboard avoidance, `FormView` -  Scroll to focused form element

### PR DESCRIPTION
**Pull request by dfeinzimer**
Closes Apollo 162

### CODE REVIEW

File: Sources/ArcGISToolkit/Components/Form/FormView.swift [hunk 1]
```diff
@@ -39,18 +39,25 @@ public struct FormView: View {
...
+                            .padding([.bottom], elementPadding)
+                        ForEach(model.visibleElements, id: \.self) { element in
+                            makeElement(element)
+                        }
                     }
                 }
             }
+            .onChange(of: model.focusedElement) { focusedElement in
+                if let focusedElement {
+                    withAnimation { scrollViewProxy.scrollTo(focusedElement, anchor: .top) }
```
Comment (mhdostal) [inline Sources/ArcGISToolkit/Components/Form/FormView.swift:58]: If the `focusedElement` is set before the keyboard appears the scroll view won't scroll:


https://github.com/Esri/arcgis-maps-sdk-swift-toolkit/assets/3998072/061260da-5c22-4ad7-b072-f633d038d8f7

Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Form/FormView.swift:58]: This should be detached from the keyboard entirely.

After looking at that specific form I think the issue is that there isn't enough content in the form for it to actually be scrollable.

I'll need to think more about how to handle that scenario.

Comment (mhdostal) [reply Sources/ArcGISToolkit/Components/Form/FormView.swift:58]: > After looking at that specific form I think the issue is that there isn't enough content in the form for it to actually be scrollable.

I'll do some more testing. I didn't think about that scenario.

File: Examples/Examples/FloatingPanelExampleView.swift [hunk 2]
```diff
@@ -38,7 +41,15 @@ struct FloatingPanelExampleView: View {
             map: dataModel.map,
             viewpoint: initialViewpoint
         )
-        .floatingPanel(selectedDetent: $selectedDetent, isPresented: isPresented) {
+        .onAttributionBarHeightChanged {
+            attributionBarHeight = $0
+        }
+        .ignoresSafeArea(.keyboard, edges: .bottom)
+        .floatingPanel(
+            attributionBarHeight: attributionBarHeight,
```
Comment (dfeinzimer) [inline Examples/Examples/FloatingPanelExampleView.swift:49]: By passing the attribution bar height to the floating panel we can prevent the overlap and center the panel right in the geo view.

File: Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanel.swift [hunk 3]
```diff
@@ -42,11 +48,14 @@ struct FloatingPanel<Content>: View where Content: View {
...
     
+    /// A Boolean value indicating whether the keyboard is open.
+    @State private var keyboardIsOpen = false
+    
     /// The latest recorded drag gesture value.
     @State private var latestDragGesture: DragGesture.Value?
     
     /// The maximum allowed height of the content.
-    @State private var maximumHeight: CGFloat = .infinity
+    @State private var maximumHeight: CGFloat = .zero
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanel.swift:58]: I encountered a few conditions where infinity was causing a purple runtime warning.

File: Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanel.swift [hunk 4]
```diff
@@ -218,24 +222,38 @@ struct FloatingPanel<Content>: View where Content: View {
...
         switch detent {
         case .summary:
-            return max(.minHeight, maximumHeight * 0.15)
+            return max(.minHeight, maximumHeight * 0.25)
         case .half:
-            return maximumHeight * 0.4
+            return maximumHeight * 0.5
         case .full:
-            return maximumHeight * (isCompact ? 0.90 : 1.0)
+            return maximumHeight
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanel.swift:229]: 25%, 50% and 100% seem more fitting.

File: Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanelModifier.swift [hunk 5]
```diff
@@ -89,12 +99,15 @@ private struct FloatingPanelModifier<PanelContent>: ViewModifier where PanelCont
...
+                    attributionBarHeight: attributionBarHeight,
                     backgroundColor: backgroundColor,
                     selectedDetent: selectedDetent,
                     isPresented: isPresented,
                     content: panelContent
                 )
-                .ignoresSafeArea(.all, edges: .bottom)
+                // When the panel is anchored to the bottom of the screen (compact) ignore the
+                // device's bottom safe area.
+                .ignoresSafeArea(.container, edges: isCompact ? .bottom : [])
```
Comment (dfeinzimer) [inline Sources/ArcGISToolkit/Components/FloatingPanel/FloatingPanelModifier.swift:110]: We had essentially opted out of the system provided keyboard avoidance by using `.all` here.

File: Sources/ArcGISToolkit/Components/Form/FormView.swift (hunk 1 above)
Comment (dfeinzimer) [reply Sources/ArcGISToolkit/Components/Form/FormView.swift:58]: Okay I've reworked this but it involved floating panel changes. Please re-test.

File: Examples/Examples/FloatingPanelExampleView.swift (hunk 2 above)
Comment (dfeinzimer) [reply Examples/Examples/FloatingPanelExampleView.swift:49]: Before:
Screenshot 2023-10-26 at 10 55 55
After:
Screenshot 2023-10-26 at 10 52 35

**Review by mhdostal — APPROVED** [review]

**Pull request merged**